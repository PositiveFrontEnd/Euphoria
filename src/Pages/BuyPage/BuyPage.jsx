import React from "react";
import { Link } from "react-router-dom";
import "../NotPage/NotPage.scss";
import { useSelector } from "react-redux";
import { selectorBasketFormValue } from "../../store/selectors";

const BuyPage = () => {
  const valueUser = useSelector(selectorBasketFormValue);
  return (
    <>
      <div className="note__page">
        <div>
          <p className="page__title"> Thanks for shopping {valueUser.name}</p>
          <p className="page__desc">Manager will call you later !</p>
          <p>
            <Link to="/"> Home</Link>
          </p>
        </div>
      </div>
    </>
  );
};
export default BuyPage;
