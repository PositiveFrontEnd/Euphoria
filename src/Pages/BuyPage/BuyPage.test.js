import React from "react";
import { render, screen } from "@testing-library/react";
import BuyPage from "./BuyPage";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

describe("Тестування компонента BuyPage", () => {
  test("Знімок", () => {
    const mockStore = configureStore();
    const initialState = {
      favorites: [],
      cardsArr: [],
      baskets: [],
      basketFormValue: {},
    };
    const store = mockStore(initialState);
    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <BuyPage />
        </BrowserRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("some text", () => {
    const mockStore = configureStore();
    const initialState = {
      favorites: [],
      cardsArr: [],
      baskets: [],
      basketFormValue: {},
    };
    const store = mockStore(initialState);
    render(
      <Provider store={store}>
        <BrowserRouter>
          <BuyPage />
        </BrowserRouter>
      </Provider>
    );
    expect(screen.getByText(/Thanks for shopping/)).toBeInTheDocument();
  });
});
