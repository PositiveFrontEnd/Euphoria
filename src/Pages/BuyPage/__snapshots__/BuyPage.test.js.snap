// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестування компонента BuyPage Знімок 1`] = `
<DocumentFragment>
  <div
    class="note__page"
  >
    <div>
      <p
        class="page__title"
      >
         Thanks for shopping 
      </p>
      <p
        class="page__desc"
      >
        Manager will call you later !
      </p>
      <p>
        <a
          href="/"
        >
           Home
        </a>
      </p>
    </div>
  </div>
</DocumentFragment>
`;
